pkg/reconciler/workload/apiexportcreate/apiexportcreate_controller.go

```diff
@@ -257,5 +312,27 @@ func (c *controller) process(ctx context.Context, key string) error {
 	klog.V(2).Infof("Creating APIBinding %s|%s", clusterName, reconcilerapiexport.TemporaryComputeServiceExportName)
 	_, err = c.kcpClusterClient.Cluster(clusterName).ApisV1alpha1().APIBindings().Create(ctx, binding, metav1.CreateOptions{})
 
+	if err != nil && !apierrors.IsAlreadyExists(err) {
+		klog.Errorf("Failed to create apibinding %s|%s: %v", clusterName, reconcilerapiexport.TemporaryComputeServiceExportName, err)
+		return err
+	}
+
+	// patch the apiexport, so we do not create the location/apibinding again even if it is deleted.
+	exportPatch := &apisv1alpha1.APIExport{
+		ObjectMeta: metav1.ObjectMeta{
+			UID:             export.UID,
+			ResourceVersion: export.ResourceVersion,
+			Annotations: map[string]string{
+				apisv1alpha1.AnnotationDefaultCreated: "",
+			},
+		},
+	}
+
+	patchData, err := json.Marshal(exportPatch)
```

Comment:
never created a merge patch like this by marshalling the object. Is that really safe? Aren't there some strange encoding artifacts that are no noops?